src/lib/pubkey/pubkey.cpp

```diff
@@ -315,7 +307,7 @@ size_t PK_Signer::signature_length() const {
    } else if(m_sig_format == Signature_Format::DerSequence) {
       // This is a large over-estimate but its easier than computing
       // the exact value
-      return m_op->signature_length() + (8 + 4 * m_parts);
+      return m_op->signature_length() + (8 + 4 * 2);
```

Comment:
These are a lot of magic numbers. Perhaps clarify their meaning by defining them as `constexpr` variables just above, or at least a comment? My best guess: 8 bytes of general DER header, 4 bytes of length field times number of signature parts.

Reply:
Basically this, though I certainly do not remember exactly how they were derived. I think this could be tightened up quite a bit especially as here we restrict the number of elements of a signature to exactly 2 which is all we really need for any signature scheme.

Reply:
Knowing there are at most 2 signature parts I just brute forced it for all group sizes, so we have an exact calculation for all EC* signatures and a smaller over-estimate for DSA groups